### PR TITLE
TDD Array quiz

### PR DESCRIPTION
**Pull request by miloofcroton**

### CODE REVIEW

File: lib/why.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+    return words.reduce((acc, curr) => {
+        
+        if(curr.split('').every(item => item.toLowerCase() !== 'y')){
+            acc.push(`Is it ${curr}?`);
+        }
+        return acc;
+    }, []);
+
+};
```
Comment (martypdx) [inline lib/why.js]: A bit of a premature optimization using reduce, consider:

```
return words
    .filter(word => word.includes('y') || word.includes('Y'))
    .map(word => `Is it ${word}?`);
```

File: test/why.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const assert = require('assert');
+const why = require('../lib/why');
+
+describe('why tests', () => {
+
+    const attempt = why(['dog', 'doggy', 'Yes', 'No']);
+
+    it('removes words with y and Y', () => {
+        assert.ok(!attempt.includes('doggy'));
+    });
```
Comment (martypdx) [inline test/why.test.js]: This test doesn't add a ton of value. If you used it initially to create functionality, remember you can delete tests

File: lib/why.js (hunk 1 above)
Comment (miloofcroton) [reply lib/why.js]: filter -> map is pretty similar reduce... good to know. Is it premature due to its inflexibility, whereas you can be a little more granular if you keep it to filter/map?

File: test/why.test.js (hunk 2 above)
Comment (miloofcroton) [reply test/why.test.js]: I do have a few questions about tests. For instance, what would you want for final commits when it comes to tests?

Is there any special logic on what you design the tests to prove? I mean, it's mostly common sense, but I was wondering if there're any principles worth mentioning.

**Pull request merged**